### PR TITLE
Update the common project with changes for supporting Turris Omnia and RDKB platforms

### PR DESCRIPTION
**Pull request by vitalybu**

### CODE REVIEW

File: common/CMakeLists.txt [hunk 1]
```diff
@@ -69,16 +80,20 @@ message(STATUS "CMAKE_PROGRAM_PATH - ${CMAKE_PROGRAM_PATH}")
 message(STATUS "CMAKE_MULTIAP_OUTPUT_DIRECTORY - ${CMAKE_MULTIAP_OUTPUT_DIRECTORY}")
 
 # Default Compiler flags
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -pthread -Wno-comment -Wno-unused-parameter -Wno-missing-field-initializers -Wno-address-of-packed-member -fPIC")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -pthread -Wno-comment -Wno-unused-parameter -Wno-missing-field-initializers -fPIC")
```
Comment (tomereli) [inline common/CMakeLists.txt:81]: Why did you remove `-Wno-address-of-packed-member` warning? @arnout added this for a reason - see #44

**Review by morantr — APPROVED** [review]

**Review by tomereli — CHANGES_REQUESTED** [review]

File: common/beerocks/bwl/common/sta_wlan_hal_types.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+struct SScanResult {
+    sMacAddr bssid;
+    uint8_t channel;
+    int8_t rssi;
+};
+
+} // namespace bwl
+
+#endif // _BWL_MON_WLAN_HAL_TYPES_H_
```
Comment (tomereli) [inline common/beerocks/bwl/common/sta_wlan_hal_types.h]: Missing empty line at end of file

File: common/beerocks/bwl/nl80211/mon_wlan_hal_nl80211.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    // Private data-members:
+private:
+    std::shared_ptr<char> m_temp_wav_value;
+};
+
+} // namespace nl80211
+} // namespace bwl
+
+#endif // _BWL_MON_WLAN_HAL_NL80211_H_
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/mon_wlan_hal_nl80211.h]: missing line break

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+    } else if (opcode == "DFS-CAC-COMPLETED") {
+        return ap_wlan_hal::Event::DFS_CAC_Completed;
+    } else if (opcode == "DFS-NOP-FINISHED") {
+        return ap_wlan_hal::Event::DFS_NOP_Finished;
+    }
+
+    return ap_wlan_hal::Event::Invalid;
+}
+
+#if 0
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: Why #if 0 ? Let's remove dead code.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 5]
```diff
@@ -0,0 +1,818 @@
...
+    char *reply;
+
+    // Send command
+    if (!wpa_ctrl_send_msg(cmd, &reply)) {
+        LOG(ERROR) << "failsafe_channel_get() failed!";
+        return false;
+    }
+
+    // Custom reply parsing
+    if (!strncmp(reply, "UNSPECIFIED", 11)) {
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: unsafe method - please use c++ comparison or safe libc.

Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: unsafe method - please use c++ comparison or safe libc.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 6]
```diff
@@ -0,0 +1,818 @@
...
+        chan = -1;
+        bw   = -1;
+    } else if (!strncmp(reply, "ACS", 3)) {
+        chan = bw = 0;
+
+    } else {
+        int freq;
+        sscanf(reply, "%d %*s %*s bandwidth=%d", &freq, &bw);
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: same here - sscanf is not safe and will not pass SDL / KW. We need prplMesh to be SDL compliant.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 7]
```diff
@@ -0,0 +1,818 @@
...
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::restricted_channels_set(char *channel_list)
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::restricted_channels_get(char *channel_list)
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp:362]: Just curious - why don't we actually use std::list for channel list? and if we pass char * then why not pass also the size of the list? potential issue here, and definitely KW issue.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 8]
```diff
@@ -0,0 +1,818 @@
...
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    memset(channel_list, 0, beerocks::message::RESTRICTED_CHANNEL_LENGTH);
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::read_acs_report()
+{
+    LOG(TRACE) << __func__ << " for interface: " << get_radio_info().iface_name;
+
+    // TODO: Get actual list of supported channels
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp:373]: Please refer to the task which should add it here. Please extend #457 and explain there all that needs to be done for finalizing nl80211 BWL.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 9]
```diff
@@ -0,0 +1,818 @@
...
+    return true;
+}
+
+bool ap_wlan_hal_nl80211::generate_connected_clients_events()
+{
+    LOG(TRACE) << __func__ << " - NOT IMPLEMENTED!";
+    return true;
+}
+
+std::string ap_wlan_hal_nl80211::get_radio_driver_version() { return "nl80211"; }
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp:444]: Shouldn't this also include a TODO which will eventually call the driver NL to get the version? This is assuming there is such an NL.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp [hunk 10]
```diff
@@ -0,0 +1,818 @@
...
+        // Initialize the message
+        memset(msg_buff.get(), 0, sizeof(sACTION_APMANAGER_CLIENT_ASSOCIATED_NOTIFICATION));
+
+        msg->params.vap_id = 0;
+        msg->params.mac    = beerocks::net::network_utils::mac_from_string(parsed_obj["_mac"]);
+
+        bool caps_valid = true;
+
+        // TODO: Update station capabilities
+#if 0
```
Comment (tomereli) [inline common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: I'm fine with a TODO, but do we need the dead code?

File: common/beerocks/bwl/CMakeLists.txt [hunk 11]
```diff
@@ -83,13 +98,50 @@ elseif(BWL_TYPE STREQUAL "WAV")
...
+    )
+
     file(GLOB bwl_platform_sources
         ${bwl_platform_sources}
         ${MODULE_PATH}/dwpal/*.c*
     )
 
-    find_package(dwpal REQUIRED)
-    list(APPEND BWL_LIBS dwpal::dwpal)
+    # find_package(dwpal REQUIRED)
```
Comment (tomereli) [inline common/beerocks/bwl/CMakeLists.txt]: Let's just remove this comment and drop the idea of having DWPAL build with CMAKE once and for all.

File: common/CMakeLists.txt (hunk 1 above)
Comment (vitalybu) [reply common/CMakeLists.txt:81]: Older GCCs don't support this flag.
Added only for GCC > 4.9...

Comment (arnout) [reply common/CMakeLists.txt:81]: > Older GCCs don't support this flag.
> Added only for GCC > 4.9...

Then use CMake's `check_cxx_compiler_flag` function to test if the flag is available. As noted in #44, the flag is *required* for recent GCC.

File: common/beerocks/bwl/common/sta_wlan_hal_types.h (hunk 2 above)
Comment (morantr) [reply common/beerocks/bwl/common/sta_wlan_hal_types.h]: done

File: common/beerocks/bwl/nl80211/mon_wlan_hal_nl80211.h (hunk 3 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/mon_wlan_hal_nl80211.h]: done

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 4 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: Ok, I am removing it. At most, we will add it back from dwpal/wav if needed.

File: common/CMakeLists.txt (hunk 1 above)
Comment (arnout) [reply common/CMakeLists.txt:81]: Never mind, it's handled a few lines below.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 5 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: done

Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: done

File: common/CMakeLists.txt [hunk 12]
```diff
@@ -12,22 +12,33 @@ endif()
 
 include(GNUInstallDirs)
 
+# Default values
+set(BEEROCKS_BRIDGE_IFACE "br-lan")
+set(BEEROCKS_PLATFORM_INIT "true")
```
Comment (arnout) [inline common/CMakeLists.txt]: Actually, I think the linux platform is the only one where this needs to be true, so it should better default to false, no?

File: common/CMakeLists.txt [hunk 13]
```diff
@@ -12,22 +12,33 @@ endif()
...
+set(BEEROCKS_INSTALL_PATH "/opt/prplmesh")
```
Comment (arnout) [inline common/CMakeLists.txt:18]: For now I'd stick to /opt/beerocks. @tomereli we discussed this in Berlin but I forgot the conclusion.

File: common/CMakeLists.txt [hunk 14]
```diff
@@ -12,22 +12,33 @@ endif()
...
 elseif (TARGET_PLATFORM STREQUAL "linux")
     add_definitions(-DBEEROCKS_LINUX)
-    set(BEEROCKS_BRIDGE_IFACE "br-lan")
     set(BEEROCKS_BH_WIRE_IFACE "sim-eth0")
+    set(BEEROCKS_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}")
     set(BEEROCKS_TMP_PATH "/tmp/$ENV{USER}/beerocks")
+elseif (TARGET_PLATFORM STREQUAL "turris-omnia")
+    set(BEEROCKS_BH_WIRE_IFACE "eth1")
+    set(BEEROCKS_PLATFORM_INIT "false")
+    set(BEEROCKS_WLAN1_IFACE "wlan0")
```
Comment (arnout) [inline common/CMakeLists.txt]: This is the default so not needed.

File: common/CMakeLists.txt [hunk 15]
```diff
@@ -69,16 +80,20 @@ message(STATUS "CMAKE_PROGRAM_PATH - ${CMAKE_PROGRAM_PATH}")
...
 if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.9)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-address-of-packed-member")
     set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fstack-protector-strong")
 else()
     set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fstack-protector")
 endif()
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CMAKE_C_FLAGS_RELEASE}")
 
+# Release flags
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -s -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Wl,-S -fPIE")
```
Comment (arnout) [inline common/CMakeLists.txt:93]: This change seems to be completely unrelated.

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 16]
```diff
@@ -149,9 +149,6 @@ start_function() {
             ;;
         * ) err "unsupported mode: $PRPLMESH_MODE"; usage; exit 1 ;;
     esac
-
-    # Give write permissions
-    chmod -R o+w /tmp/${SUDO_USER:-${USER}}
```
Comment (arnout) [inline common/beerocks/scripts/prplmesh_utils.sh.in:154]: Why is this removed? It is very useful when running on Ubuntu.

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 17]
```diff
@@ -166,12 +163,12 @@ stop_function() {
 }
 
 main_agent_operational() {
-    ps -aux | grep beerocks_agent | grep -v wlan | grep -v grep > /dev/null 2>&1 || return 1
+    @PS_CMD@ | grep beerocks_agent | grep -v wlan | grep -v grep > /dev/null 2>&1 || return 1
```
Comment (arnout) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: This and all other ps commands could use `pgrep` or `pgrep -l` instead. That also removes the need for defining `PS_CMD`.

File: common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h [hunk 18]
```diff
@@ -26,7 +26,7 @@ namespace beerocks {
 #define BEEROCKS_PLAT_MGR_UDS "uds_platform_manager"
 #define BEEROCKS_BACKHAUL_MGR_UDS "uds_backhaul_manager"
 
-#if __GNUC__ > 4
+#if __GNUC__ >= 7
 #define FALLTHROUGH [[fallthrough]]
```
Comment (arnout) [inline common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h]: Actually, if we're anyway using a define for it, and it's done in a gcc condition, we could just as well use `__attribute__((fallthrough))` which has been supported in GCC since basically forever. So the condition could just be `#if __GNUC__`.

Yes, `[[fallthrough]]` is actually standardized, but within a `#if __GNUC__` that doesn't matter much.

File: common/beerocks/bwl/common/ap_wlan_hal.h [hunk 19]
```diff
@@ -65,6 +65,41 @@ class ap_wlan_hal : public virtual base_wlan_hal {
...
+
+    /*!
+     * Disable the radio interface
+     *
+     * @return true on success or false on error.
+     */
+    virtual bool disable() = 0;
+
+    /*!
+     * Set start_disabled flag
```
Comment (arnout) [inline common/beerocks/bwl/common/ap_wlan_hal.h:83]: This should have a better documentation. Based on this explanation, I wouldn't be able to implement the function and wouldn't know when to call it.

File: common/beerocks/bwl/common/ap_wlan_hal.h [hunk 20]
```diff
@@ -261,6 +296,14 @@ class ap_wlan_hal : public virtual base_wlan_hal {
      */
     virtual bool read_acs_report() = 0;
 
+    /*!
+     * Read the supported channls from the hardware
+     * On successful completion the information can be retrieved 
+     *
+     * @return true on success or false on error.
+     */
+    virtual bool read_supported_channels() = 0;
```
Comment (arnout) [inline common/beerocks/bwl/common/ap_wlan_hal.h:305]: Why is this not just done on initialisation?

File: common/beerocks/bwl/common/ap_wlan_hal.h [hunk 21]
```diff
@@ -273,6 +316,13 @@ class ap_wlan_hal : public virtual base_wlan_hal {
      */
     virtual bool set_vap_enable(const std::string &iface_name, const bool enable) = 0;
     virtual bool get_vap_enable(const std::string &iface_name, bool &enable)      = 0;
+
+    /*!
+     * Generate "AP-STA-CONNECTED" events on connected clients on AP
+     * 
+     * @return true on success or false on error.
+     */
+    virtual bool generate_connected_clients_events() = 0;
```
Comment (arnout) [inline common/beerocks/bwl/common/ap_wlan_hal.h:325]: Why would you ever want to NOT get connected client events?

**Review by arnout — CHANGES_REQUESTED** [review]
I reviewed up to " common: bwl (common): add new interface functions ", stopping here to get the comments out.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 6 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: done

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 7 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp:362]: I don't know, but the function is not being used. Guess this is the reason no one has ever bothered to change it in other bwl's (which has the same function signature).

File: common/beerocks/bwl/CMakeLists.txt [hunk 22]
```diff
@@ -24,25 +27,37 @@ if (TARGET_PLATFORM STREQUAL "ugw")
...
 
 # Linux
 elseif(TARGET_PLATFORM STREQUAL "linux")
-    set(BWL_TYPE_DEFAULT "DUMMY")
+    set(BWL_TYPE "DUMMY")
     set(BEEROCKS_TMP_PATH "/tmp/$ENV{USER}/beerocks")
     add_definitions(-DBEEROCKS_TMP_PATH="${BEEROCKS_TMP_PATH}")
+
+# Turris Omnia
+elseif(TARGET_PLATFORM STREQUAL "turris-omnia")
```
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt:47]: Could we change this `turris-omnia` into `openwrt`? The only thing that is Turris-specific is the cross-compiler, which is not defined anywhere in the CMakeLists themselves, only in the toolchainfile.

Note that the addition of this platform has already been merged into master, so it's a global change that should probably be done in a separate PR.

File: common/beerocks/bwl/CMakeLists.txt [hunk 23]
```diff
@@ -83,13 +98,50 @@ elseif(BWL_TYPE STREQUAL "WAV")
 
 elseif(BWL_TYPE STREQUAL "DWPAL")
     
+    include_directories(
+        ${HOSTAPD_DIR}/src/drivers
+        ${PLATFORM_BUILD_DIR}/iwlwav-iw-4.14
+        ${PLATFORM_INCLUDE_DIR}/libnl3
+    )
```
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt:105]: I don't see how this change is related to "add cmake turris-omnia support" (in other words, it should have been a separate commit).

(nitpick, not really needed to change this.)

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 8 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp:373]: I'll refer to task #457, but I don't know what to write in the task (what needs to be done for finalizing nl80211 BWL).

File: common/beerocks/bwl/CMakeLists.txt [hunk 24]
```diff
@@ -83,13 +98,50 @@ elseif(BWL_TYPE STREQUAL "WAV")
...
+        -DCONFIG_CTRL_IFACE
+        -DCONFIG_CTRL_IFACE_UNIX
+    )
+
+    # Hostapd/NL80211 include directories
+    include_directories(
+        ${HOSTAPD_DIR}/src/utils
+        ${HOSTAPD_DIR}/src/common
+        ${HOSTAPD_DIR}/src/drivers
+        ${PLATFORM_INCLUDE_DIR}/libnl-tiny
```
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt:135]: This should be done with `find_package(libnl-tiny REQUIRED)`. Or whatever it is you need to do to use pkg-config.

File: common/beerocks/bwl/CMakeLists.txt [hunk 25]
```diff
@@ -83,13 +98,50 @@ elseif(BWL_TYPE STREQUAL "WAV")
...
+    )
+
+    # Platform libraries
+    link_directories(
+        ${PLATFORM_STAGING_DIR}/usr/lib
```
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt:140]: Should come out of find_package.

File: common/beerocks/bwl/CMakeLists.txt [hunk 26]
```diff
@@ -83,13 +98,50 @@ elseif(BWL_TYPE STREQUAL "WAV")
...
+    )
+
+    list(APPEND BWL_LIBS nl-tiny)
```
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt:143]: Should come out of find_package.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 10 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: removed

File: common/beerocks/bwl/CMakeLists.txt [hunk 27]
```diff
@@ -104,8 +156,9 @@ endif()
 
 # Build the library
 add_library(${PROJECT_NAME} SHARED ${bwl_common_sources} ${bwl_platform_sources})
+set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-Wl,-z,defs")
```
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt:158]: Why do we need `-Wl,-z,defs`? This warrants a comment or at least a mention in the commit message.

**Review by arnout — CHANGES_REQUESTED** [review]
Minor comments only. I actually do have major comments as well, but those should be handled as separate tasks so I won't post them here.

File: common/beerocks/bwl/CMakeLists.txt (hunk 11 above)
Comment (morantr) [reply common/beerocks/bwl/CMakeLists.txt]: done

File: common/CMakeLists.txt (hunk 12 above)
Comment (morantr) [reply common/CMakeLists.txt]: set to false by default.

File: common/CMakeLists.txt (hunk 13 above)
Comment (morantr) [reply common/CMakeLists.txt:18]: @arnout @tomereli  Let me know what you decide.

File: common/CMakeLists.txt (hunk 14 above)
Comment (morantr) [reply common/CMakeLists.txt]: removed

File: common/CMakeLists.txt (hunk 15 above)
Comment (morantr) [reply common/CMakeLists.txt:93]: @vitalybu  Please comment

File: common/beerocks/scripts/prplmesh_utils.sh.in (hunk 16 above)
Comment (morantr) [reply common/beerocks/scripts/prplmesh_utils.sh.in:154]: returned back, for now.
@vitalybu please comment.

File: common/beerocks/scripts/prplmesh_utils.sh.in (hunk 17 above)
Comment (morantr) [reply common/beerocks/scripts/prplmesh_utils.sh.in]: `pgrep -la` is better. changing.

File: common/CMakeLists.txt (hunk 15 above)
Comment (vitalybu) [reply common/CMakeLists.txt:93]: Unrelated to what?
This line was simply moved below the `if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.9)` test to include the `-Wno-address-of-packed-member` flag in case the compiler supports it...

File: common/CMakeLists.txt (hunk 13 above)
Comment (tomereli) [reply common/CMakeLists.txt:18]: @arnout I think we agreed that we'll change *only* the directory path to be /opt/prplmesh, I would go with that.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 6 above)
Comment (morantr) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: BTW @tomereli, I converted it to cpp using string stream, but it really ugly. The correct way to do it is with regex lib, but I remember that I asked in the past why don't we use it, I was answered that it was heavy or something.
So just let you know, maybe you will decide to change it.

File: common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h (hunk 18 above)
Comment (morantr) [reply common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h]: I'm not sure I understand what I suppose to do here.

File: common/beerocks/bwl/common/ap_wlan_hal.h (hunk 20 above)
Comment (morantr) [reply common/beerocks/bwl/common/ap_wlan_hal.h:305]: What is not done in init? reading the supported channels? It can't be done on init because bwl needs to be attached first.

File: common/beerocks/bwl/common/ap_wlan_hal.h (hunk 21 above)
Comment (morantr) [reply common/beerocks/bwl/common/ap_wlan_hal.h:325]: It is used in passive mode when the ap_manager is transmitting and clients are able to connect it before the agent has established the link to the controller.

File: common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h (hunk 18 above)
Comment (morantr) [reply common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h]: fixed

File: common/beerocks/bwl/CMakeLists.txt (hunk 22 above)
Comment (morantr) [reply common/beerocks/bwl/CMakeLists.txt:47]: Agreed, we will do it in a different PR, after everything will be merged to the master branch.

File: common/beerocks/bwl/CMakeLists.txt (hunk 24 above)
Comment (morantr) [reply common/beerocks/bwl/CMakeLists.txt:135]: Why?

File: common/beerocks/bwl/CMakeLists.txt (hunk 27 above)
Comment (morantr) [reply common/beerocks/bwl/CMakeLists.txt:158]: These flags force the linker to report (as errors) any undefined symbols in the shared library.

File: common/beerocks/bwl/common/ap_wlan_hal.h (hunk 19 above)
Comment (morantr) [reply common/beerocks/bwl/common/ap_wlan_hal.h:83]: It can be understood from the context.
There is a flag called "start_disabled" which this function is set on a specific vap_id. Also, it is possible to peek on the dwpal implementation to see how it implements it.

This is must be a new function since I don't recall having it in bwl, as well I didn't find any call to it on ap_manager/monitor.

@kantera800 can you add something about this function?

File: common/beerocks/bwl/CMakeLists.txt (hunk 24 above)
Comment (arnout) [reply common/beerocks/bwl/CMakeLists.txt:135]: Because `PLATFORM_INCLUDE_DIR` is a hack that makes it more difficult to integrate prplMesh in build systems and distros. Adding more hacks like this is just making things worse.

OTOH, it is already done like that in bpl, so I'm OK with keeping it like this (and fixing both together).

File: common/beerocks/bwl/CMakeLists.txt (hunk 27 above)
Comment (arnout) [reply common/beerocks/bwl/CMakeLists.txt:158]: I can read the man page :-) However, my question is: why do we suddenly need to detect undefined symbols in the shared library (which is not something that is normally an issue).

File: common/beerocks/bwl/common/ap_wlan_hal.h (hunk 20 above)
Comment (arnout) [reply common/beerocks/bwl/common/ap_wlan_hal.h:305]: > What is not done in init? reading the supported channels?

Yes, why is there a need to call this function explicitly, instead of just doing it.

> It can't be done on init because bwl needs to be attached first.

OK, then why is it not done on attach?

Anyway, I think the entire BWL should be looked at at a bit a higher level to redefine it as a clean API. So I created #477 for now, and the current situation is OK for now.

File: common/CMakeLists.txt (hunk 15 above)
Comment (arnout) [reply common/CMakeLists.txt:93]: > This line was simply moved below the `if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.9)` test

Indeed, sorry, I hadn't noticed that.

File: common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp (hunk 6 above)
Comment (vitalybu) [reply common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp]: I don't see a problem using the C++11 regex functionality

File: common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h [hunk 28]
```diff
@@ -26,12 +26,12 @@ namespace beerocks {
 #define BEEROCKS_PLAT_MGR_UDS "uds_platform_manager"
 #define BEEROCKS_BACKHAUL_MGR_UDS "uds_backhaul_manager"
 
-#if __GNUC__ >= 7
-#define FALLTHROUGH [[fallthrough]]
+#if __GNUC__
```
Comment (arnout) [inline common/beerocks/bcl/include/beerocks/bcl/beerocks_defines.h]: On second inspection: the GCC fallthrough attribute was also only introduced in GCC 7, so this makes no sense. The first version was OK.

We could improve it by also supporting C++17 (which standardizes the `[[fallthrough]]` attribute):
```
#if __GNUC__ >= 7 || __cplusplus >= 201703L

**Review by arnout — APPROVED** [review]

**Review by tomereli — APPROVED** [review]
